src/AzureAuth/CommandMain.cs

```diff
@@ -262,10 +262,43 @@ private int GetToken()
             try
             {
                 ITokenFetcher tokenFetcher = this.TokenFetcher();
-                TokenResult tokenResult = (this.tokenFetcherOptions.Scopes == null
-                    ? tokenFetcher.GetAccessTokenAsync(this.CombinedAuthMode)
-                    : tokenFetcher.GetAccessTokenAsync(this.tokenFetcherOptions.Scopes, this.CombinedAuthMode))
-                    .Result;
+                TokenResult tokenResult = null;
+
+                // When running multiple AzureAuth processes with the same resource, client, and tenant IDs,
+
+                // They may prompt many times, which is annoying and unexpected.
+                // Use Mutex to ensure that only one process can access the corresponding resource at the same time.
+                string lockName = $"Global\\{this.Resource}_{this.Client}_{this.Tenant}";
+
+                // First parameter InitiallyOwned indicated whether this lock is owned by current thread.
+                // It should be false otherwise a dead lock could occur.
+                using (Mutex mutex = new Mutex(false, lockName))
+                {
+                    try
+                    {
+                        // Wait for other session exit.
+                        mutex.WaitOne();
+                    }
+
+                    // An AbandonedMutexException could be thrown if another process exits without releasing the mutex correctly.
+                    catch (AbandonedMutexException)
+                    {
+                        // lock eventually acquired

```

Comment:
I think this comment could use just a little more clarification, but other than that things look good to me.

Reply:
I have the comment in this code block more detailed. Is that work for you?